### PR TITLE
feat(web): manually link live photos

### PR DESCRIPTION
**Pull request by jrasm91**
Add an option to manually link live photos when two assets are selected (and one is a photo and the other a video) by adding the option "Link motion video" to the menu.

Screenshot from 2024-09-09 12-23-28

### CODE REVIEW

File: web/src/lib/components/photos-page/actions/link-live-photo-action.svelte [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    clearSelect();
+    loading = false;
+  };
+</script>
+
+{#if menuItem}
+  <MenuOption {text} {icon} onClick={handleLink} />
+{/if}
+
+{#if !menuItem}
```
Comment (danieldietzler) [inline web/src/lib/components/photos-page/actions/link-live-photo-action.svelte:38]:
```suggestion
{:else}
```

**Review by danieldietzler — APPROVED** [review]

File: web/src/lib/components/photos-page/actions/link-live-photo-action.svelte (hunk 1 above)
Comment (jrasm91) [reply web/src/lib/components/photos-page/actions/link-live-photo-action.svelte:38]: nty

**Comment by pyccl** [comment]
Is this connecting when storing files or two files? Are there still two files when downloading? Is the newly downloaded file also a static image (separated from the video)?

**Comment by jrasm91** [comment]
Live photos have 2 flavors android and iOS. Android is a single file with the video embedded in the jpeg. iOS is two separate assets that are linked together. This PR is adding the option to manually link 2 separate assets together to form an iOS version of a live photo. When that version is downloaded you get 2 separate assets.

**Comment by waclaw66** [comment]
How to unlink when a mistake is made?
Video icon overlaps another icon, panorama in my case...

obrazek

**Comment by jrasm91** [comment]
Lol I guess I should have figured this would be inevitable. Why in the world did you link a panorama to a video?

**Comment by waclaw66** [comment]
> Lol I guess I should have figured this would be inevitable. Why in the world did you link a panorama to a video?

On purpose, testing edge cases ;)

**Comment by NightMean** [comment]
Thanks for this! :) If I understand correctly, this links the file only for immich but does not use the Content Identifier metadata that Apple uses on iOS. With the Content Identifier you can import the photo back to Apple Photos and retain the Live Photo as well. There is a separate project which can does this but only on a MacOS. 
https://github.com/RhetTbull/makelive

Also, it might be worth taking a look at https://github.com/PetrVys/MotionPhoto2/ which can create single HEIC live photo file that is compatible in Google Photos/iOS and of course Immich.

**Comment by koenhendriks** [comment]
Is there any automated way to do this? I've just imported ~800GB from a Google Photo's export and all the iOS live photos have become 2 separate files.

I really don't want to do this by hand 😔

**Pull request merged**